packages/dds/merge-tree/src/test/obliterate.partialLength.spec.ts

```diff
@@ -0,0 +1,368 @@
+        refSeq = client.getCurrentSeq();
+    });
+
+    afterEach(() => {
+        PartialSequenceLengths.options.verify = false;
+    });
+
+    it("removes text", () => {
+        assert.equal(client.getText(), "hello world");
+        client.obliterateRange(
+            0,
+            client.getLength(),
+            refSeq,
+            localClientId,
+            refSeq + 1,
+            false,
+            undefined as any,
+        );
+        assert.equal(client.getText(), "");
+
+        validatePartialLengths(localClientId, client.mergeTree, [
+            { seq: refSeq, len: "hello world".length },
+            { seq: refSeq + 1, len: 0 },
+        ], 0);
+        validatePartialLengths(remoteClientId, client.mergeTree, [
+            { seq: refSeq, len: "hello world".length },
+            { seq: refSeq + 1, len: 0 },
+        ]);
+    });
+
+    describe("overlapping remove+obliterate", () => {
+        it("passes for local remove and local obliterate", () => {
+            client.removeRangeLocal(
+                0,
+                "hello ".length,
+            );
+            client.obliterateRange(
+                0,
+                "hello ".length,
+                refSeq,
+                localClientId,
+                refSeq + 1,
+                false,
+                undefined as any,
+            );
+
+            validatePartialLengths(localClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "world".length },
+            ], refSeq);
+            validatePartialLengths(remoteClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "world".length },
+            ]);
+        });
+
+        it("passes for local remove and remote obliterate", () => {
+            client.removeRangeLocal(
+                0,
+                "hello ".length,
+            );
+            client.obliterateRange(
+                0,
+                "hello ".length,
+                refSeq,
+                remoteClientId,
+                refSeq + 1,
+                false,
+                undefined as any,
+            );
+
+            validatePartialLengths(localClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "world".length },
+            ]);
+            validatePartialLengths(remoteClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "world".length },
+            ], 0);
+        });
+
+        it("passes for remote remove and local obliterate", () => {
+            client.removeRangeRemote(
+                0,
+                "hello ".length,
+                refSeq + 1,
+                refSeq,
+                client.getLongClientId(remoteClientId),
+            );
+            client.obliterateRange(
```

Comment:
this approach is kind of odd in that we don't go through the normal local op application flow. I'd kind of prefer that we invoke some local variant of `obliterateRange` first, then run `removeRangeRemote`, then validate partial lengths, then ack the local obliterate, then validate partial lengths again.

Reply:
I think this bit generally applies to the tests with local ops.